### PR TITLE
Update existing and include new release action

### PR DESCRIPTION
**Pull request by harishp8889**
# Description
1. Update existing lib release action to automatically generate release notes
2. Add action to release driver and modules to build and scan repository, tag and push images, and create the github release. This action requires additional improvements where there are "TODOs" and a significant amount of testing. Please note this won't be introduced until CSM 1.13.

# Issues
List the issues impacted by this PR:

| Issue ID |
| -------- |
|     https://github.com/dell/csm/issues/1490     | 

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Ran against forked repo: https://github.com/harishp8889/test-gobrick-automation/actions/runs/11158812934/job/31015900741

### CODE REVIEW

**Comment by prablr79** [comment]
@harishp8889  can you add the Github ID for this PR ?

File: .github/workflows/csm-release-driver-module.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+      - name: Create Release
+        id: release_notes
+        uses: softprops/action-gh-release@v2
+        with:
+          tag_name: ${{ github.event.inputs.version }}
+          name: Release ${{ github.event.inputs.version }}
+          draft: true
+          prerelease: false
+          generate_release_notes: true
```
Comment (shaynafinocchiaro) [inline .github/workflows/csm-release-driver-module.yaml:67]: Does this add a changelog? Are we no longer adding the CSM changelog?

File: .github/workflows/csm-release-libs.yaml [hunk 2]
```diff
@@ -24,6 +24,12 @@ jobs:
     name: Build, Scan and Release
     runs-on: ubuntu-latest
     steps:
+      - name: Validate version input
+        run: |
+          if [[ "${{ github.event.inputs.version }}" == "none" ]]; then
```
Comment (shaynafinocchiaro) [inline .github/workflows/csm-release-libs.yaml:29]: If default is removed, this check isn't necessary.

File: .github/workflows/csm-release-driver-module.yaml (hunk 1 above)
Comment (shaynafinocchiaro) [reply .github/workflows/csm-release-driver-module.yaml:101]: Added CSM changelog and docs

File: .github/workflows/csm-release-driver-module.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+# Invocable as a reusable workflow
+on:
+  workflow_call:  
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version to release (major, minor, patch)'
+        required: true
+        default: 'none'
```
[the remark this answers is not in the document]
Comment (shaynafinocchiaro) [reply .github/workflows/csm-release-driver-module.yaml]: Removed default

File: .github/workflows/csm-release-driver-module.yaml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+jobs:
+  build-and-scan:
+    name: Build, Scan and Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Validate input
+        run: |
+          if [[ "${{ github.event.inputs.version }}" == "none" || "${{ github.event.inputs.image }}" == "none" ]]; then
+            echo "Invalid version specified: ${{ github.event.inputs.version }} Must be x.x.x"
```
[the remark this answers is not in the document]
Comment (shaynafinocchiaro) [reply .github/workflows/csm-release-driver-module.yaml]: Removed default and this check

File: .github/workflows/csm-release-driver-module.yaml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        uses: actions/checkout@v4
+
+      - name: Log in to Quay.io
+        run: echo "${{ secrets.QUAY_PASSWORD }}" | docker login quay.io -u ${{ secrets.QUAY_USERNAME }} --password-stdin # TODO we need to store quay credentials in GitHub Secrets 
+
+      - name: Pull nightly image from Quay.io
+        run: docker pull quay.io/dell/container-storage-modules/${{ github.event.inputs.image }}:nightly
+
+      - name: Retag image
+        run: docker tag quay.io/dell/container-storage-modules/${{ github.event.inputs.image }}:nightly quay.io/dell/container-storage-modules/${{ github.event.inputs.image }}:${{ github.event.inputs.version }}
```
[the remark this answers is not in the document]
Comment (shaynafinocchiaro) [reply .github/workflows/csm-release-driver-module.yaml]: Added v

File: .github/workflows/csm-release-driver-module.yaml [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+      - name: Push image to Quay.io
+        run: docker push quay.io/dell/container-storage-modules/${{ github.event.inputs.image }}:${{ github.event.inputs.version }}
```
[the remark this answers is not in the document]
Comment (shaynafinocchiaro) [reply .github/workflows/csm-release-driver-module.yaml]: Added v

**Review by ChristianAtDell — APPROVED** [review]

**Review by ChristianAtDell — APPROVED** [review]

**Review by prablr79 — APPROVED** [review]

**Pull request merged**